dbms/src/Server/Client.cpp

```diff
@@ -197,7 +197,20 @@ class Client : public Poco::Util::Application
 
             default_database = config.getString("database", "");
             user = config.getString("user", "");
-            password = config.getString("password", "");
+
+            if (config.getBool("ask-password", false))
+            {
+                std::string prompt = "Password for user " + user + ": ";
+                if (char * line_ = getpass(prompt.c_str()))
```

Comment:
This function is obsolete.  Do not use it.  If you want to read input without terminal echoing enabled, see the description of the ECHO flag in termios(3).